ofrak_core/ofrak/model/tag_model.py

```diff
@@ -1,21 +1,25 @@
 import functools
-from collections import defaultdict
-from typing import Set, Iterable, Tuple
+from typing import Iterable, Set, Tuple, Optional, cast, List
 
 
 class ResourceTag(type):
-    @functools.lru_cache(None)
+    def __init__(cls, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        cls._specificity: Optional[int] = None
+
     def tag_specificity(cls) -> int:
         """
         Indicates how specific an abstraction this tag is.
         :return: The number of classes in the inheritance hierarchy between this class and
         Resource
         """
-        specificity = 0
-        for base in cls.base_tags():
-            specificity = max(specificity, base.tag_specificity())
+        if cls._specificity is None:
+            specificity = 0
+            for base in cls.base_tags():
+                specificity = max(specificity, base.tag_specificity())
 
-        return specificity + 1
+            cls._specificity = specificity + 1
```

Comment:
Could this be a helper function that runs in __init__?

Reply:
Yep, I should have just done that. It's an easy fix.